### PR TITLE
Merged all components into one main file.

### PR DESCRIPTION
**Pull request by yorambenyaacov**
@lcaggio I have implemented @ludoo requests.

let me know if I missed anything.

### CODE REVIEW

File: data-solutions/data-platform-foundations/datasource/main.tf [hunk 1]
```diff
@@ -16,4 +16,208 @@
...
+#                                   IAM                                       #
+###############################################################################
+
+module "datamart-default-service-accounts" {
+  source     = "../../../modules/iam-service-account"
+  project_id = var.project_ids.datamart
+  name       = var.project_service_account.datamart
+
+  iam_project_roles = {
+    "${var.project_ids.datamart}" = [
```
Comment (lcaggio) [inline data-solutions/data-platform-foundations/datasource/main.tf:76]: I would refer to the project ID with a link to the module, not the variable. Please update also other elements.

Example: `module.project-datamart`.

File: data-solutions/data-platform-foundations/datasource/variables.tf [hunk 2]
```diff
@@ -12,3 +12,147 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+variable "datamart_bq_datasets" {
+  description = "Datamart Bigquery datasets"
+  type        = map(any)
+  default = {
+    bq_datamart_dataset = {
+      id       = "bq_datamart_dataset"
```
Comment (lcaggio) [inline data-solutions/data-platform-foundations/datasource/variables.tf:20]: we may want to add support for IAM roles, similar to PubSub topics

File: data-solutions/data-platform-foundations/datasource/variables.tf [hunk 3]
```diff
@@ -12,3 +12,147 @@
...
+    }
+  }
+}
+
+variable "dwh_bq_datasets" {
+  description = "DWH Bigquery datasets"
+  type        = map(any)
+  default = {
+    bq_raw_dataset = {
+      id       = "bq_raw_dataset"
```
Comment (lcaggio) [inline data-solutions/data-platform-foundations/datasource/variables.tf:31]: we may want to add support for IAM roles, similar to PubSub topics

File: data-solutions/data-platform-foundations/datasource/main.tf (hunk 1 above)
Comment (lcaggio) [reply data-solutions/data-platform-foundations/datasource/main.tf:76]: actually, do we need the `project` module?

**Review by lcaggio — COMMENTED** [review]
I would rename the `datasource` folder with `resources`

**Comment by yorambenyaacov** [comment]
@lcaggio, I'm done with the changes. Let me know what you think.

File: data-solutions/data-platform-foundations/resources/main.tf [hunk 4]
```diff
@@ -163,39 +163,34 @@ module "bucket-transformation" {
...
 
-  for_each = var.datamart_bq_datasets
+  for_each = var.dwh_bq_datasets
   id       = each.value.id
   location = each.value.location
-
-  access = {
-    owner = { role = "OWNER", type = "user" }
-  }
+  access   = each.value.access
```
Comment (lcaggio) [inline data-solutions/data-platform-foundations/resources/main.tf:188]: We need to merge `access` and `access_identities` configured in variables with the one we need from the solution to work, a logic similar to the one implemented for pubSub permissions: https://github.com/terraform-google-modules/cloud-foundation-fabric/blob/d7825f9e915901d76ff0e35e6eb580629a7286bc/data-solutions/data-platform-foundations/resources/main.tf#L20-L30

**Review by lcaggio — COMMENTED** [review]
Thanks Yoram for including changes. I left one comment. In addition, local variable `module_version` in main.tf is not used, I would remove it.

**Review by lcaggio — APPROVED** [review]
LGTM

**Comment by ludoo** [comment]
Let me do a review tonight ;)

**Pull request merged**